payroll-processor-client/src/app/employee/employee-list/employee-list.component.ts

```diff
@@ -1,30 +1,38 @@
-import { Component, OnInit, OnDestroy } from '@angular/core';
-import { Employee } from '../../data/state/employee-model';
-import { DataService } from 'src/app/data/data-service';
-import { faSmileBeam, faSkull } from '@fortawesome/free-solid-svg-icons';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+
+import { faSkull, faSmileBeam } from '@fortawesome/free-solid-svg-icons';
+
+import { Employee } from './state/employee-list.model';
+import { EmployeeListQuery } from './state/employee-list.query';
+import { EmployeeListService } from './state/employee-list.service';
 
 @Component({
   selector: 'app-employee-list',
   templateUrl: './employee-list.component.html',
   styleUrls: ['./employee-list.component.scss'],
 })
 export class EmployeeListComponent implements OnInit, OnDestroy {
-  faSkull = faSkull;
-  faSmileBeam = faSmileBeam;
+  readonly faSkull = faSkull;
+  readonly faSmileBeam = faSmileBeam;
+  readonly employees: Observable<Employee[]>;
 
-  constructor(private dataService: DataService) {}
+  constructor(
+    private service: EmployeeListService,
+    private query: EmployeeListQuery,
+  ) {
+    this.employees = this.query.selectAll();
```

Comment:
If you want to get super cool - you can assign the query directly to the field!

```typescript
  readonly faSkull = faSkull;
  readonly faSmileBeam = faSmileBeam;
  readonly employees = this.query.selectAll();
```

Reply:
Eventually what we'll end up doing is creating projections of state inside the query class that are combinations of other projections of state:

```
// query.ts

readonly activeEmployeeNames = this.selectActive(({ firstName, lastName }) => `${firstName} ${lastName}`);

/// component.ts

readonly activeEmployeeNames = this.query.activeEmployeeNames;
```

This lets us encapsulate mapping/projection/aggregation logic inside the Query class, and remove a lot of boilerplate in the component class - no explicit field type, no assignment in constructor, no future mutations of the field.

Reply:
this makes sense since we don't use the query object anywhere else in this class